lib/glific/triggers.ex

```diff
@@ -199,11 +205,18 @@ defmodule Glific.Triggers do
     {:ok, entered_flow} =
       Repo.fetch_by(FlowRevision, %{flow_id: flow.id, status: "published"})
 
-    entered_first_node = entered_flow |> Map.get(:definition) |> Map.get("nodes") |> hd()
-    entered_action = hd(entered_first_node["actions"])
-    entered_type = Map.get(entered_action, "type")
+    entered_action =
+      entered_flow |> Map.get(:definition) |> Map.get("nodes") |> hd() |> Map.get("actions")
 
-    handle_message_type(entered_type, entered_action, attrs)
+    case entered_action do
+      [] ->
+        do_create_trigger(attrs)
+
+      [entered_action | _rest] ->
+        entered_type = Map.get(entered_action, "type")
+
+        handle_message_type(entered_type, entered_action, attrs)
+    end
```

Comment:
This looks similar to the above logic. Extract the above logic in a function and use in both places.